skimage/measure/fit.py

```diff
@@ -32,16 +38,64 @@ def from_estimate(cls, *data): ...
     def residuals(self, *data): ...
 
 
+_PARAMS_DEP_START = '0.26'
+_PARAMS_DEP_STOP = '2.2'
+
+
 class BaseModel:
-    def __init__(self):
-        self.params = None
+    def __init_subclass__(self):
```

Comment:
Off-topic: I guess this works to discourage subclassing `BaseModel`. 

We could also flat out refactor to `_fit.py` and deprecate any access to `skimage.measure.fit`. But I'm thinking that's something we could just do during the transition to skimage2. Maybe with a some meta-programming shenanigans that warns for all access to modules that are not exposed in our HTML documentation (synonymous with the tree defined by `__all__`).